### PR TITLE
[REEF-1027] Remove old dependencies from Tang by making Tint project

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
This PR removes old reflections/guava/javassist dependencies from Tang
by splitting Tint as a separate project.

JIRA:
  [REEF-1027](https://issues.apache.org/jira/browse/REEF-1027)

Pull request:
  This closes #

### CODE REVIEW

File: lang/java/reef-common/src/main/java/org/apache/reef/util/DeadlockInfo.java [hunk 1]
```diff
@@ -18,7 +18,8 @@
  */
 package org.apache.reef.util;
 
-import javax.annotation.Nullable;
+import org.apache.avro.reflect.Nullable;
```
Comment (markusweimer) [inline lang/java/reef-common/src/main/java/org/apache/reef/util/DeadlockInfo.java]: Why this change?

Comment (dongjoon-hyun) [inline lang/java/reef-common/src/main/java/org/apache/reef/util/DeadlockInfo.java]: Oh, It's my mistake. I will add JSR305 dependency.

**Comment by dongjoon-hyun** [comment]
The failure, `WatcherTest.testEventStream`, on one of Travis CI  is not relevant to this PR.

**Comment by markusweimer** [comment]
I will test and merge.

**Comment by dongjoon-hyun** [comment]
Thank you @markusweimer !
